--- src/core/plugins/players/plugins_list.h	2018-02-19 23:24:41.000000000 +0100
+++ src/core/plugins/players/plugins_list.h	2024-09-21 11:33:59.493194047 +0200
@@ -44,20 +44,20 @@
   void RegisterFTCSupport(PlayerPluginsRegistrator& registrator);
   void RegisterCOPSupport(PlayerPluginsRegistrator& registrator);
   void RegisterTFESupport(PlayerPluginsRegistrator& registrator);
-  void RegisterXMPPlugins(PlayerPluginsRegistrator& registrator);
-  void RegisterSIDPlugins(PlayerPluginsRegistrator& registrator);
-  void RegisterET1Support(PlayerPluginsRegistrator& registrator);
-  void RegisterAYCSupport(PlayerPluginsRegistrator& registrator);
-  void RegisterSPCSupport(PlayerPluginsRegistrator& registrator);
-  void RegisterMTCSupport(PlayerPluginsRegistrator& registrator);
-  void RegisterGMEPlugins(PlayerPluginsRegistrator& registrator);
-  void RegisterAHXSupport(PlayerPluginsRegistrator& registrator);
-  void RegisterPSFSupport(PlayerPluginsRegistrator& registrator);
-  void RegisterUSFSupport(PlayerPluginsRegistrator& registrator);
-  void RegisterGSFSupport(PlayerPluginsRegistrator& registrator);
-  void Register2SFSupport(PlayerPluginsRegistrator& registrator);
-  void RegisterSDSFSupport(PlayerPluginsRegistrator& registrator);
-  void RegisterASAPPlugins(PlayerPluginsRegistrator& registrator);
+  // void RegisterXMPPlugins(PlayerPluginsRegistrator& registrator);
+  // void RegisterSIDPlugins(PlayerPluginsRegistrator& registrator);
+  // void RegisterET1Support(PlayerPluginsRegistrator& registrator);
+  // void RegisterAYCSupport(PlayerPluginsRegistrator& registrator);
+  // void RegisterSPCSupport(PlayerPluginsRegistrator& registrator);
+  // void RegisterMTCSupport(PlayerPluginsRegistrator& registrator);
+  // void RegisterGMEPlugins(PlayerPluginsRegistrator& registrator);
+  // void RegisterAHXSupport(PlayerPluginsRegistrator& registrator);
+  // void RegisterPSFSupport(PlayerPluginsRegistrator& registrator);
+  // void RegisterUSFSupport(PlayerPluginsRegistrator& registrator);
+  // void RegisterGSFSupport(PlayerPluginsRegistrator& registrator);
+  // void Register2SFSupport(PlayerPluginsRegistrator& registrator);
+  // void RegisterSDSFSupport(PlayerPluginsRegistrator& registrator);
+  // void RegisterASAPPlugins(PlayerPluginsRegistrator& registrator);
 
   void RegisterPlayerPlugins(PlayerPluginsRegistrator& registrator)
   {
@@ -91,19 +91,19 @@
     RegisterFTCSupport(registrator);
     RegisterCOPSupport(registrator);
     RegisterTFESupport(registrator);
-    RegisterXMPPlugins(registrator);
-    RegisterSIDPlugins(registrator);
-    RegisterET1Support(registrator);
-    RegisterAYCSupport(registrator);
-    RegisterSPCSupport(registrator);
-    RegisterMTCSupport(registrator);
-    RegisterGMEPlugins(registrator);
-    RegisterAHXSupport(registrator);
-    RegisterPSFSupport(registrator);
-    RegisterUSFSupport(registrator);
-    RegisterGSFSupport(registrator);
-    Register2SFSupport(registrator);
-    RegisterSDSFSupport(registrator);
-    RegisterASAPPlugins(registrator);
+    // RegisterXMPPlugins(registrator);
+    // RegisterSIDPlugins(registrator);
+    // RegisterET1Support(registrator);
+    // RegisterAYCSupport(registrator);
+    // RegisterSPCSupport(registrator);
+    // RegisterMTCSupport(registrator);
+    // RegisterGMEPlugins(registrator);
+    // RegisterAHXSupport(registrator);
+    // RegisterPSFSupport(registrator);
+    // RegisterUSFSupport(registrator);
+    // RegisterGSFSupport(registrator);
+    // Register2SFSupport(registrator);
+    // RegisterSDSFSupport(registrator);
+    // RegisterASAPPlugins(registrator);
   }
 }
