--- apps/libzxtune/src/zxtune.cpp	2018-02-19 23:24:41.000000000 +0100
+++ apps/libzxtune/src/zxtune.cpp	2024-09-21 11:33:59.493194047 +0200
@@ -443,3 +443,19 @@
     return false;
   }
 }
+std::string ZXTune_GetInfo(ZXTuneHandle player, std::string paramName)
+{
+    const PlayerWrapper::Ptr wrapper = PlayersCache::Instance().Get(player);
+    const Parameters::Accessor::Ptr props = wrapper->GetParameters();
+    String res;
+    props->FindValue(paramName, res);
+    return res;
+}
+ZXTUNE_API long ZXTune_GetDuration(ZXTuneHandle player)
+{
+    const PlayerWrapper::Ptr wrapper = PlayersCache::Instance().Get(player);
+    const Parameters::Accessor::Ptr props = wrapper->GetParameters();
+    Parameters::IntType frameDuration = Parameters::ZXTune::Sound::FRAMEDURATION_DEFAULT;
+    props->FindValue(Parameters::ZXTune::Sound::FRAMEDURATION, frameDuration);
+    return frameDuration;
+}
\ No newline at end of file
