--- src/Config.h.in	2024-10-27 14:19:49.000000000 +0100
+++ src/Config.h.in	2025-01-06 16:08:34.568530935 +0100
@@ -71,16 +71,16 @@
 #undef PACKAGE_VERSION
 
 /* The size of 'char', as computed by sizeof. */
-#undef SIZEOF_CHAR
+#define SIZEOF_CHAR @SIZEOF_CHAR@
 
 /* The size of 'int', as computed by sizeof. */
-#undef SIZEOF_INT
+#define SIZEOF_INT @SIZEOF_INT@
 
 /* The size of 'long int', as computed by sizeof. */
-#undef SIZEOF_LONG_INT
+#define SIZEOF_LONG_INT @SIZEOF_LONG_INT@
 
 /* The size of 'short int', as computed by sizeof. */
-#undef SIZEOF_SHORT_INT
+#define SIZEOF_SHORT_INT @SIZEOF_SHORT_INT@
 
 /* Define to 1 if all of the C89 standard headers exist (not just the ones
    required in a freestanding environment). This macro is provided for
