--- src/FC.cpp	2010-06-30 00:06:36.000000000 +0200
+++ src/FC.cpp	2024-09-21 11:33:59.196531170 +0200
@@ -1163,3 +1163,38 @@
     }
     CHXdata.period = tmp0;
 }
+
+int FC::getUsedPatterns()
+{
+	return _admin.usedPatterns;
+}
+
+int FC::getUsedSndModSeqs()
+{
+	return _admin.usedSndModSeqs;
+}
+
+int FC::getUsedVolModSeqs()
+{
+	return _admin.usedVolModSeqs;
+}
+
+unsigned short FC::getSampleLength(unsigned int qual)
+{
+	return _sounds[qual].len*2;
+}
+
+unsigned short FC::getSampleRepOffset(unsigned int qual)
+{
+	return _sounds[qual].repOffs;
+}
+
+unsigned short FC::getSampleRepLength(unsigned int qual)
+{
+	return _sounds[qual].repLen*2;
+}
+
+void FC::muteChannel(unsigned int qual)
+{
+    _CHdata[qual].volume=0;
+}
