--- src/FC.h	2010-06-30 00:04:41.000000000 +0200
+++ src/FC.h	2024-09-21 11:33:59.196531170 +0200
@@ -29,6 +29,15 @@
     FC();
     ~FC();
 
+	
+	int getUsedPatterns();
+	int getUsedSndModSeqs();
+	int getUsedVolModSeqs();
+	unsigned short getSampleLength(unsigned int qual);
+	unsigned short getSampleRepOffset(unsigned int qual);
+	unsigned short getSampleRepLength(unsigned int qual);
+    void muteChannel(unsigned int qual);
+
     void setMixer(PaulaMixer*);
     bool isOurData(void*,unsigned long int);
     bool init(void*,udword,int=0,int=0);
@@ -109,6 +118,7 @@
             udword silence;
         } offsets;
 
+		int numPatterns;
         int usedPatterns;
         int usedSndModSeqs;
         int usedVolModSeqs;
