--- src/fc14audiodecoder.h	2008-10-19 20:30:00.000000000 +0200
+++ src/fc14audiodecoder.h	2024-09-21 11:33:59.196531170 +0200
@@ -66,6 +66,15 @@
     /* Fill output sample buffer with audio. */
     void fc14dec_buffer_fill(void* decoder, void* buffer, unsigned long int length);
 
+	int fc14dec_get_used_patterns(void* decoder);
+	int fc14dec_get_used_snd_mod_seqs(void* decoder);
+	int fc14dec_get_used_vol_mod_seqs(void* decoder);
+	int fc14dec_get_sample_length(void* decoder, unsigned int qual);
+	int fc14dec_get_sample_rep_offset(void* decoder, unsigned int qual);
+	int fc14dec_get_sample_rep_length(void* decoder, unsigned int qual);
+    void fc14dec_mute_channel(void* ptr, bool mute, unsigned int qual);
+    unsigned short int fc14dec_get_channel_volume(void* ptr, unsigned int qual);
+
 #ifdef __cplusplus
 }
 #endif
