--- release/distrib/replay/lib/include/endianness.h	2024-02-05 23:48:28.000000000 +0100
+++ release/distrib/replay/lib/include/endianness.h	2024-09-26 12:41:44.791226864 +0200
@@ -35,7 +35,7 @@
 #if defined(__WIN32__) && !defined(__GCC__)
 typedef unsigned __int64 Uint64;
 #else
-#if defined(__HAIKU__) || defined(__LINUX__)
+#if defined(__HAIKU__) || defined(__LINUX__) || defined(BZR2)
 #include <stdint.h>
 typedef uint64_t Uint64;
 #else
