--- src/files/include/files_list.h	2025-04-10 08:43:29.318185000 +0200
+++ src/files/include/files_list.h	2025-04-10 08:46:02.549454000 +0200
@@ -34,10 +34,12 @@
 
 // ------------------------------------------------------
 // Includes
-#if defined(__GCC__)
+#if defined(BZR2)
+#include "../include/variables.h"
+#elif defined(__GCC__)
 #include "../../include/variables.h"
 #include "../../ui/include/misc_draw.h"
-#elif !defined(BZR2)
+#else
 #include "../include/variables.h"
 #include "../ui/include/misc_draw.h"
 #endif
