--- src/files/include/mods.h	2024-02-05 23:48:28.000000000 +0100
+++ src/files/include/mods.h	2024-09-21 11:33:59.333196091 +0200
@@ -37,8 +37,8 @@
 #include "files.h"
 #include "reverbs.h"
 #include "../../editors/include/patterns_blocks.h"
-#include "../../midi/include/midi.h"
-#include "../../ui/include/misc_draw.h"
+//#include "../../midi/include/midi.h"
+//#include "../../ui/include/misc_draw.h"
 
 // ------------------------------------------------------
 // Structures
