--- release/distrib/replay/lib/include/replay.h	2024-12-24 19:47:28.592674094 +0100
+++ release/distrib/replay/lib/include/replay.h	2024-12-24 19:59:16.340277784 +0100
@@ -37,31 +37,7 @@
 #include "tb_303.h"
 #include "cubic_spline.h"
 #include "spline.h"
-#if defined(__WIN32__)
-#include "../sounddriver/include/sounddriver_windows.h"
-#elif defined(__FREEBSD__)
-#include "../sounddriver/include/sounddriver_freebsd.h"
-#elif defined(__NETBSD__)
-#include "../sounddriver/include/sounddriver_netbsd.h"
-#elif defined(__LINUX__)
-#include "../sounddriver/include/sounddriver_linux.h"
-#elif defined(__MACOSX_PPC__)
-#include "../sounddriver/include/sounddriver_macosx.h"
-#elif defined(__MACOSX_X86__)
-#include "../sounddriver/include/sounddriver_macosx.h"
-#elif defined(__AROS__) || defined(__MORPHOS__)
-#include "../sounddriver/include/sounddriver_aros.h"
-#elif defined(__AMIGAOS4__)
-#include "../sounddriver/include/sounddriver_aos4.h"
-#elif defined(__PSP__)
-#include "../sounddriver/include/sounddriver_psp.h"
-#elif defined(__PSVITA__)
-#include "../sounddriver/include/sounddriver_psvita.h"
-#elif defined(__HAIKU__)
-#include "../sounddriver/include/sounddriver_haiku.h"
-#else
-#error "No sound driver defined !"
-#endif
+#include "sounddriver_dummy.h"
 #include "samples_unpack.h"
 #include "ptkreplay.h"
 #include "synth.h"
