--- release/distrib/replay/lib/replay.cpp	2024-12-24 20:07:28.517589156 +0100
+++ release/distrib/replay/lib/replay.cpp	2024-12-24 20:11:59.040959669 +0100
@@ -955,7 +955,7 @@
 void STDCALL Mixer(Uint8 *Buffer, Uint32 Len)
 {
 
-#if defined(__MACOSX_PPC__) || defined(__MACOSX_X86__)
+#if defined(__MACOSX_PPC__) || defined(__MACOSX_X86__) || defined(BZR2)
     float *pSamples_flt = (float *) Buffer;
 #endif
 
@@ -967,6 +967,7 @@
     float clamp_right_value;
 #endif

+Uint32 numSamples = Len;
 #if !defined(__STAND_ALONE__) && !defined(__WINAMP__)
     if(thread_sema)
     {
@@ -981,10 +982,21 @@
 
 #if defined(__MACOSX_PPC__) || defined(__MACOSX_X86__)
         for(i = Len - 1; i >= 0; i -= 8)
+#elif defined(BZR2)
+        for (; numSamples; --numSamples)
 #else
         for(i = Len - 1; i >= 0; i -= 4)
 #endif
         {
+            if (!Song_Playing)
+                break;
+            if (done)
+            {
+                if (numSamples == Len)
+                    done = FALSE;
+                break;
+            }
+
             Get_Player_Values();
 
 #if !defined(__STAND_ALONE__)
@@ -1024,7 +1036,7 @@
 #endif
 #endif
 
-#if defined(__MACOSX_PPC__) || defined(__MACOSX_X86__)
+#if defined(__MACOSX_PPC__) || defined(__MACOSX_X86__) || defined (BZR2)
             *pSamples_flt++ = left_float;
             *pSamples_flt++ = right_float;
 #else
