--- src/files/reverbs.cpp	2024-02-05 23:48:28.000000000 +0100
+++ src/files/reverbs.cpp	2024-09-21 11:33:59.333196091 +0200
@@ -35,9 +35,9 @@
 
 // ------------------------------------------------------
 // Load the data from a reverb file (or a module)
-void Load_Reverb_Data(int (*Read_Function)(void *, int ,int, FILE *),
-                      int (*Read_Function_Swap)(void *, int ,int, FILE *),
-                      FILE *in, int New)
+void Load_Reverb_Data(int (*Read_Function)(void *, int ,int, ReplayerFile&),
+                      int (*Read_Function_Swap)(void *, int ,int, ReplayerFile&),
+                      ReplayerFile& in, int New)
 {
     int i;
 
