--- src/files/include/synths.h	2025-01-27 21:00:24.894516050 +0100
+++ src/files/include/synths.h	2025-01-27 21:00:24.887849461 +0100
@@ -46,7 +46,7 @@
                        int (*Read_Function_Swap)(void *, int ,int, FILE *),
                        FILE *in,
 #else
-void Read_Synth_Params(int (*Read_Function)(void *, int ,int, FILE *),
+void Read_Synth_Params(int (*Read_Function)(void *, int ,int, CustomFile &),
                        int (*Read_Function_Swap)(void *, int ,int, CustomFile &),
                        CustomFile &in,
 #endif
