--- src/include/variables.h	2024-02-05 23:48:28.000000000 +0100
+++ src/include/variables.h	2024-09-26 12:41:44.794560150 +0200
@@ -36,9 +36,9 @@
 
 // ------------------------------------------------------
 // Includes
-#include <SDL/SDL.h>
+//#include <SDL/SDL.h>
 
-#include "../support/include/main.h"
+//#include "../support/include/main.h"
 #include "../../release/distrib/replay/lib/include/replay.h"
 
 // ------------------------------------------------------
@@ -375,21 +375,27 @@
 extern int resty;
 extern int rs_coef;
 
+#if !defined(BZR2)
 extern SDL_Surface *Main_Screen;
+#endif
 extern int CONSOLE_HEIGHT2;
 
 extern float left_float_render;
 extern float right_float_render;
 
+#if !defined(BZR2)
 extern SDL_Surface *PFONT;
 extern SDL_Surface *FONT;
 extern SDL_Surface *FONT_LOW;
+#endif
 
 extern int gui_action;
 extern int gui_action_metronome;
 extern char teac;
 
+#if !defined(BZR2)
 extern MOUSE Mouse;
+#endif
 
 extern int fluzy;
 
@@ -438,8 +444,10 @@
 extern float *Scope_Dats[MAX_TRACKS];
 extern float *Scope_Dats_LeftRight[2];
 
+#if !defined(BZR2)
 extern SDL_Surface *SKIN303;
 extern SDL_Surface *LOGOPIC;
+#endif
 
 extern int MouseWheel_Multiplier;
 extern char Rows_Decimal;
