--- src/common.h	2025-01-01 01:11:02.000000000 +0100
+++ src/common.h	2025-01-02 15:15:35.288619702 +0100
@@ -26,6 +26,7 @@
 #include <stdlib.h>
 #include <stdio.h>
 #include <string.h>
+#include "queue.h"
 #include "xmp.h"
 
 #undef  LIBXMP_EXPORT_VAR
@@ -556,7 +557,12 @@
 
 	double current_time;
 	double frame_time;
-
+Queue patternPosBuffer;
+Queue trackPosBuffer;
+Queue posBuffer;
+Queue vumeterBuffer;
+Queue speedBuffer;
+Queue bpmBuffer;
 	int loop_count;
 	int sequence;
 	unsigned char sequence_control[XMP_MAX_MOD_LENGTH];
