--- src/loaders/prowizard/pm01.c	2021-06-10 02:24:24.000000000 +0200
+++ src/loaders/prowizard/pm01.c	2024-12-23 10:43:16.860567916 +0100
@@ -7,6 +7,7 @@
  */
 
 #include <math.h>
+#include "period.h"
 #include "prowiz.h"
 
 static int depack_pm01(HIO_HANDLE *in, FILE *out)
@@ -115,7 +116,7 @@
 			note = (int)((double)note *
 					pow(2, -1.0 * fin[j % 4] / 12 / 8));
 			if (note > 0) {
-				note = period_to_note(note) - 48;
+				note = libxmp_period_to_note(note) - 48;
 			}
 
 			p[0] = ptk_table[note][0];
