--- include/xmp.h	2025-01-01 01:11:02.000000000 +0100
+++ include/xmp.h	2025-01-02 13:28:59.929886000 +0100
@@ -399,6 +400,7 @@
 LIBXMP_EXPORT int         xmp_smix_channel_pan (xmp_context, int, int);
 LIBXMP_EXPORT int         xmp_smix_load_sample (xmp_context, int, const char *);
 LIBXMP_EXPORT int         xmp_smix_release_sample (xmp_context, int);
+LIBXMP_EXPORT void        xmp_get_channel_volumes  (xmp_context, unsigned char*);
 
 #ifdef __cplusplus
 }
