--- AtariAudio/SndhFile.cpp	2026-08-30 01:00:52.000000000 +0200
+++ AtariAudio/SndhFile.cpp	2026-08-31 20:11:32.005000000 +0200
@@ -28,6 +28,8 @@
 	m_rawBuffer = nullptr;
 	m_Title = nullptr;
 	m_Author = nullptr;
+	m_Ripper = nullptr;
+	m_Converter = nullptr;
 	m_sYear = nullptr;
 	m_rawSize = 0;
 	m_playerRate = 0;
@@ -121,9 +123,14 @@
 					m_Author = read8 + 4;
 					read8 = skipNTString(read8 + 4);
 				}
-				else if (	(0 == strncmp(read8, "RIPP", 4)) ||
-							(0 == strncmp(read8, "CONV", 4)))
+				else if (0 == strncmp(read8, "RIPP", 4))
 				{
+					m_Ripper = read8 + 4;
+					read8 = skipNTString(read8 + 4);
+				}
+				else if (0 == strncmp(read8, "CONV", 4))
+				{
+				    m_Converter = read8 + 4;
 					read8 = skipNTString(read8 + 4);
 				}
 				else if ((0 == strncmp(read8, "YEAR", 4)))
@@ -219,6 +226,8 @@
 	out.samplePerTick = m_hostReplayRate / m_playerRate;
 	out.musicName = m_Title;
 	out.musicAuthor = m_Author;
+	out.ripper = m_Ripper;
+	out.converter = m_Converter;
 	out.year = m_sYear;
 
 	out.subsongCount = m_subSongCount;
