--- AtariAudio/SndhFile.h	2024-03-22 00:57:02.000000000 +0100
+++ AtariAudio/SndhFile.h	2024-09-21 11:33:59.346529254 +0200
@@ -24,6 +24,8 @@
 		int samplePerTick;
 		const char* musicName;
 		const char* musicAuthor;
+        const char*	ripper;
+        const char*	converter;
 		const char* year;
 	};
 
@@ -57,6 +59,8 @@
 	bool	m_bLoaded;
 	char*	m_Title;
 	char*	m_Author;
+    char*	m_Ripper;
+    char*	m_Converter;
 	char*	m_sYear;
 	const void*	m_rawBuffer;
 	int		m_rawSize;
