--- AtariAudio/SndhFile.h	2026-08-30 01:00:52.000000000 +0200
+++ AtariAudio/SndhFile.h	2026-08-31 19:57:16.781000000 +0200
@@ -24,6 +24,8 @@
 		int samplePerTick;
 		const char* musicName;
 		const char* musicAuthor;
+		const char*	ripper;
+		const char*	converter;
 		const char* year;
 	};
 
@@ -58,6 +60,8 @@
 	bool	m_bLoaded;
 	const char*	m_Title;
 	const char*	m_Author;
+	const char* m_Ripper;
+	const char* m_Converter;
 	const char*	m_sYear;
 	const void*	m_rawBuffer;
 	int		m_rawSize;
