--- libavformat/demux.c	2026-01-08 18:08:30.421557000 +0100
+++ libavformat/demux.c	2026-01-08 20:32:59.403755754 +0100
@@ -308,11 +308,12 @@
             goto fail;
         }
 
-    if (!s->metadata) {
+    if (si->id3v2_meta) {
+        if (s->metadata) {
+            av_log(s, AV_LOG_WARNING, "Discarding ID3 tags because more suitable tags were found.\n");
+        }
         s->metadata    = si->id3v2_meta;
         si->id3v2_meta = NULL;
-    } else if (si->id3v2_meta) {
-        av_log(s, AV_LOG_WARNING, "Discarding ID3 tags because more suitable tags were found.\n");
         av_dict_free(&si->id3v2_meta);
     }
 
