--- libopenmpt/libopenmpt.hpp	2023-12-13 08:52:44.000000000 +0100
+++ libopenmpt/libopenmpt.hpp	2024-09-21 11:33:59.203197752 +0200
@@ -808,6 +808,13 @@
 	  \return Metadata item keys supported by openmpt::module::get_metadata
 	  \sa openmpt::module::get_metadata
 	*/
+	
+	int get_mod_sample_size(int32_t index) const;
+	int get_mod_sample_loopstart(int32_t index) const;
+	int get_mod_sample_loopend(int32_t index) const;
+	int get_mod_sample_volume(int32_t index) const;
+	int get_mod_sample_finetune(int32_t index) const;
+	
 	LIBOPENMPT_CXX_API_MEMBER std::vector<std::string> get_metadata_keys() const;
 	//! Get a metadata item value
 	/*!
