--- libopenmpt/libopenmpt.h	2023-12-13 08:52:44.000000000 +0100
+++ libopenmpt/libopenmpt.h	2024-09-21 11:33:59.203197752 +0200
@@ -1235,6 +1235,12 @@
  */
 LIBOPENMPT_API float openmpt_module_get_current_channel_vu_rear_right( openmpt_module * mod, int32_t channel );
 
+LIBOPENMPT_API int32_t get_mod_sample_size(int32_t index);
+LIBOPENMPT_API int32_t get_mod_sample_loopstart(int32_t index);
+LIBOPENMPT_API int32_t get_mod_sample_loopend(int32_t index);
+LIBOPENMPT_API int32_t get_mod_sample_volume(int32_t index);
+LIBOPENMPT_API int32_t get_mod_sample_finetune(int32_t index);
+
 /*! \brief Get the number of sub-songs
  *
  * \param mod The module handle to work on.
