--- soundlib/Sndfile.h	2024-01-10 21:33:43.000000000 +0100
+++ soundlib/Sndfile.h	2024-09-21 11:33:59.203197752 +0200
@@ -816,6 +816,11 @@
 	std::string GetTitle() const { return m_songName; }
 	bool SetTitle(const std::string &newTitle); // Return true if title was changed.
 	const char *GetSampleName(SAMPLEINDEX nSample) const;
+	SmpLength GetModSampleSize(SAMPLEINDEX nSample) const;
+	SmpLength GetModSampleLoopStart(SAMPLEINDEX nSample) const;
+	SmpLength GetModSampleLoopEnd(SAMPLEINDEX nSample) const;
+	uint16 GetModSampleVolume(SAMPLEINDEX nSample) const;
+	uint8 GetModSampleFineTune(SAMPLEINDEX nSample) const;
 	const char *GetInstrumentName(INSTRUMENTINDEX nInstr) const;
 	uint32 GetMusicSpeed() const { return m_PlayState.m_nMusicSpeed; }
 	TEMPO GetMusicTempo() const { return m_PlayState.m_nMusicTempo; }
