--- src/snd/music.c	2020-09-26 08:03:28.000000000 +0200
+++ src/snd/music.c	2025-11-26 21:22:49.487475665 +0100
@@ -96,7 +96,14 @@
 
 static int RWread(struct RWops *context, void *ptr, int size, int maxnum)
 {
-	return fread(ptr, size, maxnum, context->fp);
+    // grabbed from RWread function in ksnd.c
+
+	const int len = my_min(size * maxnum, context->mem.length - context->mem.ptr);
+	memcpy(ptr, context->mem.base + context->mem.ptr, len);
+
+	context->mem.ptr += len;
+
+	return len;
 }
 
 
