--- src/uade.c	2019-06-21 15:52:56.000000000 +0200
+++ src/uade.c	2025-02-26 14:50:04.554401862 +0100
@@ -47,6 +47,7 @@
 #include "amigamsg.h"
 #include "ossupport.h"
 #include "sysincludes.h"
+#include <winsock2.h> // for ntohl etc.
 
 
 enum print_help {
@@ -60,7 +61,6 @@
 #ifndef EMSCRIPTEN
 static 
 #endif
-void change_subsong(int subsong);
 
 static int uade_calc_reloc_size(uae_u32 *src, uae_u32 *end);
 static int uade_get_u32(int addr);
@@ -110,6 +110,8 @@
 int uade_read_size;
 int uade_reboot;
 int uade_time_critical;
+int max_subsongs;
+int min_subsongs;
 
 static int disable_modulechange;
 static int old_ledstate;
@@ -129,6 +131,14 @@
 static char epoptions[256];
 static size_t epoptionsize;
 
+struct uade_state* get_uade_state(void)
+{
+  return &_state;
+}
+void set_uade_state(struct uade_state* state)
+{
+  loadsettings(state);
+}
 
 static void add_ep_option(const char *s)
 {
@@ -226,8 +236,38 @@
   return ((int) offset);
 }
 
+
+int get_quit()
+{
+  return quit_program;
+}
+
+int get_silence_detected()
+{
+  return silence_detected;
+}
+
+int get_missing_file()
+{
+  return missing_file;
+}
+
 #ifdef EMSCRIPTEN
-struct uade_sample_data sample_data= {.alloclen=0, .buflen= 0, .buf= 0, .is_new = 0};
+struct uade_sample_data sample_data= {0, 0, 0, 0};
+
+int get_samples(void* buffer)
+{
+  if (sample_data.is_new != 0)
+  {
+    memcpy((char *)buffer, sample_data.buf, sample_data.buflen);
+    //printf("buflen: %i\n",sample_data.buflen);
+    //fflush(stdout);
+    sample_data.is_new=0;
+    return 1;
+  }
+  return 0;
+}
+
 struct uade_sample_data * get_new_samples() {
 	if (sample_data.is_new != 0) {
 		return &sample_data;
@@ -268,6 +308,7 @@
 	if (uade_test_silence(sndbuffer, bytes, &_state)) {
 	  fprintf(stderr, "silence detected (%d seconds)\n", _state.config.silence_timeout);
 		quit_program = 1;
+	    silence_detected = 1;
 	}
   
   // signal that we have a new buffer full of sample data.. somebody should better go and fetch it..
@@ -355,6 +396,16 @@
 char _meta_data_ready= 0;
 #endif
 
+int get_max_subsongs()
+{
+  return max_subsongs;
+}
+
+int get_min_subsongs()
+{
+  return min_subsongs;
+}
+
 void uade_get_amiga_message(void)
 {
   uae_u8 *ptr;
@@ -404,6 +455,9 @@
 	snprintf(inf_maxs, sizeof inf_maxs, "%d", maxs);
 	snprintf(inf_curs, sizeof inf_curs, "%d", curs);
 
+    max_subsongs = maxs;
+    min_subsongs = mins;
+
 //fprintf(stderr, "amiga song info: %d %d %d\n", mins, curs, maxs);
 	
 	// easier to use JavaScript to parse that info_text crap
@@ -546,6 +600,7 @@
 #ifndef EMSCRIPTEN
     if ((file = uade_open_amiga_file((char *) nameptr, uade_player_dir))) {
 #else
+    fprintf (stderr, "AMIGAMSG_LOADFILE: (%s)\n", nameptr);
 	uo= uade_open_amiga_file((char *) nameptr, uade_player_dir);
     if ((file = uo->file)) {
 #endif
@@ -554,9 +609,12 @@
       fclose(file); file = NULL;
       uade_put_long(0x20C, len);
       uade_send_debug("load success: %s ptr 0x%x size 0x%x", nameptr, dst, len);
+      fprintf (stderr, "load success: %s ptr 0x%x size: %i\n", nameptr, dst, len);
     } else {
 		// testcase: AYPlayers/ZXAYEMUL is indirectly loaded from players/PlayAY
       uade_send_debug("load: file not found: %s", nameptr);
+      fprintf (stderr, "load: file not found: %s\n", nameptr);
+      uade_put_long(0x20C, len);
     }
     break;
 
@@ -573,6 +631,7 @@
 #ifndef EMSCRIPTEN
     if ((file = uade_open_amiga_file((char *) nameptr, uade_player_dir))) {
 #else
+    fprintf (stderr, "AMIGAMSG_READ: (%s)\n", nameptr);
 	uo= uade_open_amiga_file((char *) nameptr, uade_player_dir);
     if ((file = uo->file)) {
 #endif
@@ -603,6 +662,7 @@
 #ifndef EMSCRIPTEN
     if ((file = uade_open_amiga_file((char *) nameptr, uade_player_dir))) {
 #else
+    fprintf (stderr, "AMIGAMSG_FILESIZE: (%s)\n", nameptr);
 	uo= uade_open_amiga_file((char *) nameptr, uade_player_dir);
     if ((file = uo->file)) {
 #endif
@@ -834,7 +894,11 @@
 #ifdef EMSCRIPTEN
   snprintf(optionsfile, PATH_MAX, "%s/uaerc", basedir);
   FILE *f= uade_fopen(optionsfile, "r")->file;
-  if (f == 0) return -1;	// this file must exist in a consistent installation
+  if (f == 0)
+  {
+    fprintf(stderr, "uadecore: Could not load uaerc (%s).\n", optionsfile);
+    return -1;	// this file must exist in a consistent installation
+  }
 
     if (cfgfile_load_file (&currprefs, f) == 0) {
       fprintf(stderr, "uadecore: Could not load uaerc (%s).\n", optionsfile);
@@ -1199,6 +1263,7 @@
 
   bytesread = uade_safe_load(vaddr, file, maxlen);
   fclose(file);
+  fprintf(stderr, "file closed: %s\n", name);
   return bytesread;
 }
 
@@ -1215,7 +1280,9 @@
 	// initial start sequence (base for uade_reset)
 	uade_reboot = 1;
 	quit_program = 0;
-	
+    silence_detected = 0;
+    missing_file = 0;
+
 	if (uade_main(basedir) < 0) return -1;	
 	/*
 	* until here the setup should correspond to what m68k_go did before loop - one loop would then 
@@ -1294,7 +1361,7 @@
 
 #ifdef EMSCRIPTEN  
 	snprintf(song.scorename, sizeof song.scorename, "%s/amigasrc/score/score", basedir);
-	snprintf(song.modulename, sizeof song.modulename, "%s/%s", basedir, songmodule);
+    snprintf(song.modulename, sizeof song.modulename, "%s", songmodule);
 	snprintf(song.playername, sizeof song.playername, "");
 	//fprintf(stderr, "song.modulename: %s [%s] %s\n", song.modulename, basedir, songmodule);
 	// resolve the needed player (for "custom" modules the module becomes the player)
@@ -1394,6 +1461,7 @@
 
   /* load the module if available */
   if (song.modulename[0]) {
+    fprintf (stderr, "song.modulename: %s\n", song.modulename);
     bytesread = uade_safe_load_name(modaddr, song.modulename, "module", uade_highmem - modaddr);
 #ifdef EMSCRIPTEN
 	if (bytesread <1) { 
@@ -1576,6 +1644,16 @@
 #endif
 }
 
+void loadsettings(struct uade_state* state)
+{
+  ems_song_initialization(state);
+  m68k_reset ();
+  customreset ();	// reset of the Amiga custom-chips
+
+  uade_reboot = 0;
+  max_subsongs = 0;
+  min_subsongs = 0;
+}
 
 static void uade_put_long(int addr, int val)
 {
@@ -1702,7 +1780,7 @@
   }
 #else
   quit_program= 1;	// end and let client decide how he wants to proceed
-
+  fprintf(stderr, "quit_program uade_song_end\n");
   fprintf(stderr, "%s ended: %s\n", strrchr(song.modulename,'/'), reason);
 #endif
   /* if audio_output is zero (and thus the client is waiting for the first
@@ -1739,3 +1817,11 @@
   fprintf(stderr, "uadecore: Invalid string at 0x%x.\n", address);
   return 0;
 }
+
+void uade_notify_song_update(const char *info_text, const char *inf_mins, const char *inf_maxs, const char *inf_curs)
+{
+  song_info[0] = info_text;
+  song_info[1] = inf_mins;
+  song_info[2] = inf_maxs;
+  song_info[3] = inf_curs;
+}
