--- src/include/uae.h	2014-09-16 19:11:24.000000000 +0200
+++ src/include/uae.h	2024-09-21 11:33:59.423194942 +0200
@@ -20,7 +20,7 @@
 extern int uade_is_exit(void);
 extern void m68k_reset (void);
 extern void customreset (void);
-	
+extern void m68k_run_1 (void);
 int uade_is_reboot(void);
 int uade_is_exit(void);
 
@@ -32,6 +32,8 @@
 extern void uae_quit (void);
 
 extern int quit_program;
+extern int silence_detected;
+extern int missing_file;
 
 extern char warning_buffer[256];
 
