--- src/include/newcpu.h	2014-09-16 19:11:24.000000000 +0200
+++ src/include/newcpu.h	2024-09-21 11:33:59.423194942 +0200
@@ -47,7 +47,8 @@
 
 extern int broken_in;
 
-typedef unsigned long cpuop_func (uae_u32) REGPARAM;
+//typedef unsigned long cpuop_func (uae_u32) REGPARAM;
+typedef unsigned long cpuop_func (uae_u32);
 
 struct cputbl {
     cpuop_func *handler;
@@ -55,7 +56,8 @@
     uae_u16 opcode;
 };
 
-extern unsigned long op_illg (uae_u32) REGPARAM;
+//extern unsigned long op_illg (uae_u32) REGPARAM;
+extern unsigned long op_illg (uae_u32);
 
 typedef char flagtype;
 
