--- src/include/uade.h	2014-09-16 19:11:24.000000000 +0200
+++ src/include/uade.h	2024-09-21 11:33:59.423194942 +0200
@@ -38,8 +38,11 @@
 };
 
 
-struct uade_sample_data sample_data;
-
+//struct uade_sample_data sample_data;
+int get_samples(void*);
+int get_quit();
+int get_silence_detected();
+int get_missing_file();
 struct uade_sample_data *get_new_samples(void);
 
 #endif
@@ -50,6 +53,11 @@
 void uade_get_amiga_message(void);
 void uade_handle_r_state(void);
 #ifdef EMSCRIPTEN
+struct uade_state _state; // was in uade.c
+struct uade_state* get_uade_state(void);
+void set_uade_state(struct uade_state* state);
+int get_max_subsongs();
+int get_min_subsongs();
 void change_subsong(int subsong);
 int uade_option(const char *basedir);
 int uade_reset(int,char*,char*);
