--- cpp/lobby_manager.cpp	2022-04-27 00:13:22.000000000 +0200
+++ cpp/lobby_manager.cpp	2024-10-18 09:01:49.460681531 +0200
@@ -167,7 +167,7 @@
                                std::function<void(Result, Lobby const&)> callback)
 {
     static auto wrapper =
-      [](void* callbackData, EDiscordResult result, DiscordLobby* lobby) -> void {
+      [] DISCORD_CALLBACK (void* callbackData, EDiscordResult result, DiscordLobby* lobby) -> void {
         std::unique_ptr<std::function<void(Result, Lobby const&)>> cb(
           reinterpret_cast<std::function<void(Result, Lobby const&)>*>(callbackData));
         if (!cb || !(*cb)) {
@@ -185,7 +185,7 @@
                                LobbyTransaction const& transaction,
                                std::function<void(Result)> callback)
 {
-    static auto wrapper = [](void* callbackData, EDiscordResult result) -> void {
+    static auto wrapper = [] DISCORD_CALLBACK (void* callbackData, EDiscordResult result) -> void {
         std::unique_ptr<std::function<void(Result)>> cb(
           reinterpret_cast<std::function<void(Result)>*>(callbackData));
         if (!cb || !(*cb)) {
@@ -204,7 +204,7 @@
 
 void LobbyManager::DeleteLobby(LobbyId lobbyId, std::function<void(Result)> callback)
 {
-    static auto wrapper = [](void* callbackData, EDiscordResult result) -> void {
+    static auto wrapper = [] DISCORD_CALLBACK (void* callbackData, EDiscordResult result) -> void {
         std::unique_ptr<std::function<void(Result)>> cb(
           reinterpret_cast<std::function<void(Result)>*>(callbackData));
         if (!cb || !(*cb)) {
@@ -222,7 +222,7 @@
                                 std::function<void(Result, Lobby const&)> callback)
 {
     static auto wrapper =
-      [](void* callbackData, EDiscordResult result, DiscordLobby* lobby) -> void {
+      [] DISCORD_CALLBACK (void* callbackData, EDiscordResult result, DiscordLobby* lobby) -> void {
         std::unique_ptr<std::function<void(Result, Lobby const&)>> cb(
           reinterpret_cast<std::function<void(Result, Lobby const&)>*>(callbackData));
         if (!cb || !(*cb)) {
@@ -240,7 +240,7 @@
   std::function<void(Result, Lobby const&)> callback)
 {
     static auto wrapper =
-      [](void* callbackData, EDiscordResult result, DiscordLobby* lobby) -> void {
+      [] DISCORD_CALLBACK (void* callbackData, EDiscordResult result, DiscordLobby* lobby) -> void {
         std::unique_ptr<std::function<void(Result, Lobby const&)>> cb(
           reinterpret_cast<std::function<void(Result, Lobby const&)>*>(callbackData));
         if (!cb || !(*cb)) {
@@ -256,7 +256,7 @@
 
 void LobbyManager::DisconnectLobby(LobbyId lobbyId, std::function<void(Result)> callback)
 {
-    static auto wrapper = [](void* callbackData, EDiscordResult result) -> void {
+    static auto wrapper = [] DISCORD_CALLBACK (void* callbackData, EDiscordResult result) -> void {
         std::unique_ptr<std::function<void(Result)>> cb(
           reinterpret_cast<std::function<void(Result)>*>(callbackData));
         if (!cb || !(*cb)) {
@@ -403,7 +403,7 @@
                                 LobbyMemberTransaction const& transaction,
                                 std::function<void(Result)> callback)
 {
-    static auto wrapper = [](void* callbackData, EDiscordResult result) -> void {
+    static auto wrapper = [] DISCORD_CALLBACK (void* callbackData, EDiscordResult result) -> void {
         std::unique_ptr<std::function<void(Result)>> cb(
           reinterpret_cast<std::function<void(Result)>*>(callbackData));
         if (!cb || !(*cb)) {
@@ -426,7 +426,7 @@
                                     std::uint32_t dataLength,
                                     std::function<void(Result)> callback)
 {
-    static auto wrapper = [](void* callbackData, EDiscordResult result) -> void {
+    static auto wrapper = [] DISCORD_CALLBACK (void* callbackData, EDiscordResult result) -> void {
         std::unique_ptr<std::function<void(Result)>> cb(
           reinterpret_cast<std::function<void(Result)>*>(callbackData));
         if (!cb || !(*cb)) {
@@ -452,7 +452,7 @@
 
 void LobbyManager::Search(LobbySearchQuery const& query, std::function<void(Result)> callback)
 {
-    static auto wrapper = [](void* callbackData, EDiscordResult result) -> void {
+    static auto wrapper = [] DISCORD_CALLBACK (void* callbackData, EDiscordResult result) -> void {
         std::unique_ptr<std::function<void(Result)>> cb(
           reinterpret_cast<std::function<void(Result)>*>(callbackData));
         if (!cb || !(*cb)) {
@@ -487,7 +487,7 @@
 
 void LobbyManager::ConnectVoice(LobbyId lobbyId, std::function<void(Result)> callback)
 {
-    static auto wrapper = [](void* callbackData, EDiscordResult result) -> void {
+    static auto wrapper = [] DISCORD_CALLBACK (void* callbackData, EDiscordResult result) -> void {
         std::unique_ptr<std::function<void(Result)>> cb(
           reinterpret_cast<std::function<void(Result)>*>(callbackData));
         if (!cb || !(*cb)) {
@@ -502,7 +502,7 @@
 
 void LobbyManager::DisconnectVoice(LobbyId lobbyId, std::function<void(Result)> callback)
 {
-    static auto wrapper = [](void* callbackData, EDiscordResult result) -> void {
+    static auto wrapper = [] DISCORD_CALLBACK (void* callbackData, EDiscordResult result) -> void {
         std::unique_ptr<std::function<void(Result)>> cb(
           reinterpret_cast<std::function<void(Result)>*>(callbackData));
         if (!cb || !(*cb)) {
