--- src/coding/coding.h	2026-01-25 17:20:56.000000000 +0100
+++ src/coding/coding.h	2026-02-03 14:51:13.503550232 +0100
@@ -8,6 +8,8 @@
 //todo remove
 #include "libs/clhca.h"
 
+#include "libavutil/dict.h"
+
 /* adx_decoder */
 void decode_adx(VGMSTREAMCHANNEL* stream, sample_t* outbuf, int channelspacing, int32_t first_sample, int32_t samples_to_do, int32_t frame_bytes, coding_t coding_type, uint32_t codec_config);
 void adx_next_key(VGMSTREAMCHANNEL* stream);
@@ -602,6 +604,8 @@
 void ffmpeg_set_allow_pcm24(ffmpeg_codec_data* data);
 const char* ffmpeg_get_metadata_value(ffmpeg_codec_data* data, const char* key);
 
+const AVDictionaryEntry* ffmpeg_get_all_metadata (ffmpeg_codec_data* data);
+
 int32_t ffmpeg_get_samples(ffmpeg_codec_data* data);
 int ffmpeg_get_sample_rate(ffmpeg_codec_data* data);
 int ffmpeg_get_channels(ffmpeg_codec_data* data);
