--- src/meta/ffmpeg.c	2025-09-28 21:13:32.000000000 +0200
+++ src/meta/ffmpeg.c	2025-12-16 14:02:42.545332925 +0100
@@ -31,7 +31,7 @@
     //    return NULL;
 
     /* don't try to open headers and other mini files */
-    if (get_streamfile_size(sf) <= 0x1000)
+    if (get_streamfile_size(sf) <= 0x20)
         return NULL;
 
     /* reject some formats handled elsewhere (better fail and check there than let buggy FFmpeg take over) */
