--- src/coding/ffmpeg_decoder.c	2026-01-25 17:20:56.000000000 +0100
+++ src/coding/ffmpeg_decoder.c	2026-02-03 14:48:07.844963735 +0100
@@ -1115,6 +1115,38 @@
     return avde->value;
 }
 
+const AVDictionaryEntry* ffmpeg_get_all_metadata(ffmpeg_codec_data* data) {
+
+    static AVDictionaryEntry tags[256] = {};
+    memset(tags, 0, sizeof(AVDictionaryEntry) * 256);
+
+    if (!data || !data->formatCtx)
+        return tags;
+
+    const AVDictionaryEntry *avde = NULL;
+    int i = 0;
+    const AVDictionary *avd = data->formatCtx->metadata; /* per format (like Flac) */
+
+    if (avd) {
+        while ((avde = av_dict_iterate(avd, avde))) {
+            tags[i++] = *avde;
+        }
+    }
+
+    if (!data->formatCtx->streams)
+        return tags;
+
+    avd = data->formatCtx->streams[data->stream_index]->metadata; /* per stream (like Ogg) */
+
+    if (avd) {
+        while ((avde = av_dict_iterate(avd, avde))) {
+            tags[i++] = *avde;
+        }
+    }
+
+    return tags;
+}
+
 int32_t ffmpeg_get_samples(ffmpeg_codec_data* data) {
     if (!data)
         return 0;
