--- src/engine/engine.h	2025-06-12 10:23:59.000000000 +0200
+++ src/engine/engine.h	2025-10-15 08:04:44.757030297 +0200
@@ -447,6 +447,7 @@
   bool playing;
   bool freelance;
   bool shallStop, shallStopSched;
+public:
   bool endOfSong;
   bool consoleMode;
   bool disableStatusOut;
@@ -475,6 +476,7 @@
   int midiOutTimeRate;
   float midiVolExp;
   int softLockCount;
+public:
   int subticks, ticks, curRow, curOrder, prevRow, prevOrder, remainingLoops, totalLoops, lastLoopPos, exportLoopCount, curExportChan, nextSpeed, prevSpeed, elapsedBars, elapsedBeats, curSpeed;
   size_t curSubSongIndex;
   size_t bufferPos;
