--- src/log.cpp	2024-05-02 03:00:07.000000000 +0200
+++ src/log.cpp	2024-09-21 11:33:59.163198263 +0200
@@ -121,11 +121,13 @@
   logEntries[pos].ready=true;
 
   // write to log file
-  if (logFileAvail) {
-    appendLogBuf(logEntries[pos]);
-    logFileNotify.notify_one();
-  }
+  //
+  // if (logFileAvail) {
+  //   appendLogBuf(logEntries[pos]);
+  //   logFileNotify.notify_one();
+  // }
 
+logLevel=LOGLEVEL_INFO;
   if (logLevel<level) return 0;
   switch (level) {
     case LOGLEVEL_ERROR:
