--- src/a2m-v2.cpp	2026-02-03 17:46:00.163421765 +0100
+++ src/a2m-v2.cpp	2026-02-03 17:46:00.174421619 +0100
@@ -28,7 +28,7 @@
  * http://www.adlibtracker.net/
  *
  */
-
+// A2M-V2: Fix crash when setting volume for the channel with unset instrument (ff2rebel.a2m) (adplug commit 8b77c64)
 #include "a2m-v2.h"
 #include "debug.h"
 #include <climits>
@@ -735,6 +735,10 @@
     }
 
     tINSTR_DATA *instr = get_instr_data_by_ch(chan);
+    if (!instr) {
+        AdPlug_LogWrite("set_ins_volume: instr not set for channel %d\n", chan);
+        return;
+    }
 
     // ** OPL3 emulation workaround **
     // force muted instrument volume with missing channel ADSR data
@@ -792,6 +796,10 @@
 void Ca2mv2Player::set_volume(uint8_t modulator, uint8_t carrier, uint8_t chan)
 {
     tINSTR_DATA *instr = get_instr_data_by_ch(chan);
+    if (!instr) {
+        AdPlug_LogWrite("set_volume: instr not set for channel %d\n", chan);
+        return;
+    }
 
     // ** OPL3 emulation workaround **
     // force muted instrument volume with missing channel ADSR data
@@ -870,7 +878,10 @@
 void Ca2mv2Player::reset_ins_volume(int chan)
 {
     tINSTR_DATA *instr = get_instr_data_by_ch(chan);
-    if (!instr) return;
+    if (!instr) {
+        AdPlug_LogWrite("reset_ins_volume: instr not set for channel %d\n", chan);
+        return;
+    }
 
     uint8_t vol_mod = instr->fm.volM;
     uint8_t vol_car = instr->fm.volC;
@@ -891,6 +902,10 @@
             set_ins_volume_4op(BYTE_NULL, chan);
         } else if (ch->carrier_vol[chan] || ch->modulator_vol[chan]) {
             tINSTR_DATA *instr = get_instr_data_by_ch(chan);
+            if (!instr) {
+                AdPlug_LogWrite("set_global_volume: instr not set for channel %d\n", chan);
+                return;
+            }
 
             set_ins_volume(instr->fm.connect ? ch->fmpar_table[chan].volM : BYTE_NULL, ch->fmpar_table[chan].volC, chan);
         }
@@ -1182,7 +1197,6 @@
 
 void Ca2mv2Player::process_effects(tADTRACK2_EVENT *event, int slot, int chan)
 {
-    tINSTR_DATA *instr = get_instr_data_by_ch(chan);
     uint8_t def = event->eff[slot].def;
     uint8_t val = event->eff[slot].val;
 
@@ -1413,6 +1427,13 @@
         break;
 
     case ef_SetInsVolume:
+    {
+        tINSTR_DATA *instr = get_instr_data_by_ch(chan);
+        if (!instr) {
+            AdPlug_LogWrite("ef_SetInsVolume: instr not set for channel %d\n", chan);
+            break;
+        }
+
         if (_4op_vol_valid_chan(chan)) {
             set_ins_volume_4op(63 - val, chan);
         } else if (percussion_mode && ((chan >= 16) && (chan <= 19))) { //  in [17..20]
@@ -1423,8 +1444,16 @@
             set_ins_volume(63 - val, 63 - val, chan);
         }
         break;
+    }
 
     case ef_ForceInsVolume:
+    {
+        tINSTR_DATA *instr = get_instr_data_by_ch(chan);
+        if (!instr) {
+            AdPlug_LogWrite("ef_ForceInsVolume: instr not set for channel %d\n", chan);
+            break;
+        }
+
         if (percussion_mode && ((chan >= 16) && (chan <= 19))) { //  in [17..20]
             set_ins_volume(63 - val, BYTE_NULL, chan);
         } else if (instr->fm.connect == 0) {
@@ -1433,6 +1462,7 @@
             set_ins_volume(63 - val, 63 - val, chan);
         }
         break;
+    }
 
     case ef_PositionJump:
         if (no_loop(chan, current_line)) {
@@ -2322,8 +2352,12 @@
 inline int Ca2mv2Player::chanvol(int chan)
 {
     tINSTR_DATA *instr = get_instr_data_by_ch(chan);
+    if (!instr) {
+        AdPlug_LogWrite("chanvol: instr not set for channel %d\n", chan);
+    }
+    char instr_fm_connect = instr ? instr->fm.connect : 0;
 
-    if (instr->fm.connect == 0)
+    if (instr_fm_connect == 0)
         return 63 - ch->fmpar_table[chan].volC;
     else
         return 63 - (ch->fmpar_table[chan].volM + ch->fmpar_table[chan].volC) / 2;
