--- sidid.c	2025-09-20 12:52:04.821852652 +0200
+++ sidid.c	2025-09-20 14:06:59.005491946 +0200
@@ -543,3 +543,46 @@
   if ((c >= 'A') && (c <= 'F')) return c - 'A' + 10;
   return -1;
 }
+
+char *identifyBufferFromConfig(const char *sididCfgPath, unsigned char *buffer, int length)
+{
+  readconfig((char *) sididCfgPath);
+
+  SIDID *id;
+  int found = 0;
+
+  if (!playerid) id = firstid;
+  else id = playerid;
+
+  while (id)
+  {
+    if (identifybuffer(id, buffer, length))
+    {
+      id->count++;
+      if (!found)
+      {
+        found = 1;
+        identified++;
+      }
+
+      if (!multiscan) break;
+    }
+    if (!playerid)
+    {
+      id = (SIDID *)id->next;
+    }
+    else
+    {
+      break;
+    }
+  }
+  if (!found)
+  {
+
+    unidentified++;
+    return "Unknown";
+  }
+  examined++;
+  return id->name;
+
+}
